### ISSUE TITLE
Feature Request: Add option to change pool hostname

### ISSUE THREAD

**Issue by haslersn** [minio/minio#14843]
**Is your feature request related to a problem? Please describe.**

When changing the hostname of the server(s) of a pool, it must be changed in the MinIO command line accordingly. Currently, when doing so, MinIO will not start and instead give an error. For instance, when renaming hostnames from `original{1..4}` to `renamed{1..4}`, MinIO gives an error like this:

```
pool(1st) = https://original{1...4}.example.com:9000/disk{1...4} is not specified, please specify on server command line
```

(See also #14756.)

**Describe the solution you'd like**

It would be nice to have a flag or command line option to omit the check that checks the pools' names and order. (Only the number of pools and their size, i.e. number of servers and drives, should still be checked.) By default, this flag would be disabled. If one wants to rename hostnames, then the steps are as follows:

1. Stop MinIO.
2. Enable the flag to omit the check.
3. Start MinIO with modified command line according to the renamed hostnames.
4. (MinIO will now write the new pools' names to `pool.bin`.)
5. Disable the flag to restore the default behavior.

**Comment by harshavardhana** [comment]
That is something in the future that we plan to do, but it is a bit harder since we need to make sure that previous and newer endpoints follow the same pattern, etc.

Currently this is not possible and we will work on it when we find free cycles. Not a priority at the moment.

**Comment by harshavardhana** [comment]
> (See also #14756.)

This has nothing to do with this feature request. Just FYI.

**Comment by haslersn** [comment]
Is there a workaround? Can I perhaps remove `pool.bin`? Is that safe? How to do it? Simply stop MinIO and then `rm -rf .minio.sys/pool.bin/` on every disk?

**Comment by harshavardhana** [comment]
> Is there a workaround? Can I perhaps remove `pool.bin`? Is that safe? How to do it? Simply stop MinIO and then `rm -rf .minio.sys/pool.bin/` on every disk?

You can remove `pool.bin` and then restart with the same settings with different hostname.

**Comment by harshavardhana** [comment]
This is supported now after #16071
